impeller/typographer/backends/skia/text_render_context_skia.cc

```diff
@@ -98,9 +96,48 @@ static size_t PairsFitInAtlasOfSize(const FontGlyphPair::Vector& pairs,
   return 0;
 }
 
+static bool CanAppendToExistingAtlas(
+    std::shared_ptr<GlyphAtlas> atlas,
+    const FontGlyphPair::Vector& extra_pairs,
+    std::vector<Rect>& glyph_positions,
+    ISize atlas_size,
+    std::shared_ptr<GrRectanizer> rect_packer) {
+  TRACE_EVENT0("impeller", __FUNCTION__);
+  if (!rect_packer || atlas_size.IsEmpty()) {
+    return false;
+  }
+
+  // We assume that all existing glyphs will fit. After all, they fit before.
+  // The glyph_positions only contains the values for the additional glyphs
+  // from extra_pairs.
+  glyph_positions.clear();
+  glyph_positions.reserve(extra_pairs.size());
```

Comment:
Why pass this in at all? It seems strange to have this method clearing a vector it was handed.

Reply:
I see now.

Perhaps instead of clearing it, just DCHECK that it's empty?

Reply:
Done